x/twap/logic_test.go

```diff
@@ -961,7 +961,7 @@ func (s *TestSuite) TestUpdateRecords() {
 				// The new record added.
 				{
 					spotPriceA:    sdk.OneDec(),
-					spotPriceB:    sdk.ZeroDec(),
+					spotPriceB:    baseRecord.P1LastSpotPrice,       // getSpotPrice failed, should equal to previous record LastSpotPrice
 					lastErrorTime: baseRecord.Time.Add(time.Second), // equals to block time
```

Comment:
I think we need further test cases, with having more variant in scenarios. 

Can we add test cases where
1. getting last spot prices errored, so we use last spot price from the previous record
2. getting last spot prices errored,and getting previous record also errored, resulting in exp spot price to be zero value?